src/main/java/org/assertj/core/api/AbstractIterableAssert.java

```diff
@@ -2804,6 +2804,105 @@ private ELEMENT_ASSERT internalElement(int index) {
+   * </ul>
+   * <p>
+   * Example: default {@code Object} assertions
+   * <pre><code class='java'> List&lt;String&gt; babySimpsons = asList("Maggie");
+   *
+   * // assertion succeeds, only Object assertions are available after singleElement()
+   * assertThat(babySimpsons).singleElement()
+   *                    .isEqualTo("Maggie");
+   *
+   * // assertion fails
+   * assertThat(babySimpsons).singleElement()
+   *                    .isEqualTo("Homer");                   
+   *
+   * // assertion fails because list contains no elements
+   * assertThat(emptyList()).singleElement();
+   *                    
+   *
+   * // assertion fails because list contains more than one element
+   *  List&lt;String&gt; simpsons = asList("Homer", "Marge", "Lisa", "Bart", "Maggie");
+   * assertThat(simpsons).singleElement();</code></pre>
+   * <p>
+   * If you have created the Iterable assertion using an {@link AssertFactory} or the element assert class,
+   * you will be able to chain {@code singleElement()} with more specific typed assertion.
+   * <p>
+   * Example: use of {@code String} assertions after {@code singleElement()}
+   * <pre><code class='java'> List&lt;String&gt; babySimpsons = asList("Maggie");
+   *
+   * // assertion succeeds
+   * // String assertions are available after singleElement()
+   * assertThat(babySimpsons, StringAssert.class).singleElement()
+   *                                        .startsWith("Mag");
+   * 
+   * // assertion fails
+   * assertThat(babySimpsons, StringAssert.class).singleElement()
+   *                                        .startsWith("Lis");</code></pre>
+   *
+   * @return the assertion on the first element
+   * @throws AssertionError if the actual {@link Iterable} does not contain exactly one element.
+   * @since 3.17.0
+   * @see #singleElement(InstanceOfAssertFactory)
+   */
+  @CheckReturnValue
+  public ELEMENT_ASSERT singleElement() {
+    return hasSize(1).first();
+  }
+
+  /**
+   * Verifies that the {@link Iterable} under test contains a single element and allow to perform assertions that element.
+   * <p>
+   * This is a shorthand for <code>hasSize(1).first(assertFactory)</code>.
+   * <p>
+   * The {@code assertFactory} parameter allows to specify an {@link InstanceOfAssertFactory}, which is used to get the
+   * assertions narrowed to the factory type.
+   * <p>
+   * Example: use of {@code String} assertions after {@code singleElement(as(InstanceOfAssertFactories.STRING)}
+   * <pre><code class='java'> List&lt;String&gt; babySimpsons = asList("Maggie");
+   *
+   * // assertion succeeds
+   * assertThat(babySimpsons).singleElement(as(InstanceOfAssertFactories.STRING))
+   *                    .startsWith("Mag");
+   * 
+   * // assertion fails
+   * assertThat(babySimpsons).singleElement(as(InstanceOfAssertFactories.STRING))
```

Comment:
on this one only, use `STRING` only just to show that with importing  statically `InstanceOfAssertFactories.STRING` makes the code to read nicely